src/data.ts

```diff
@@ -147,10 +147,13 @@ export function isGraticuleGenerator(data: Partial<Data> | Partial<VgData>): dat
   return 'graticule' in data;
 }
 
-export type DataSourceType = 'raw' | 'main' | 'row' | 'column' | 'lookup';
-
-export const MAIN = 'main' as const;
-export const RAW = 'raw' as const;
+export enum DataSourceType {
+  Raw = 'raw',
+  Main = 'main',
+  Row = 'row',
+  Column = 'column',
+  Lookup = 'lookup'
+}
```

Comment:
I don't think we even need the names here anymore. Let's just let ts figure out the enum values (probably numbers).